### DISCUSSION TITLE
Proposal to Add `hasJoin` Method to Eloquent's Builder Class

### DISCUSSION THREAD

**Discussion by stojkov-z** [laravel/framework#49436] [Ideas]
Hi Everyone,
I'd like to propose the addition of a `hasJoin` method to the `Illuminate\Database\Eloquent\Builder` class. 

This method would enhance query building by allowing developers to easily check if a join to a specific table already exists in the query.

Here's an example of the `hasJoin` method I've used in one of my projects, implemented as a macro:
```
Builder::macro('hasJoin', function ($table) {
    return collect($this->getQuery()->joins)->contains(function ($join) use ($table) {
        return $join->table === $table;
    });
});
```
This macro adds a convenient hasJoin method to check if a join with a specific table exists in the query. It’s a practical solution I found useful in complex query scenarios.

I'm keen to hear your thoughts and suggestions for improvement or optimization.
Thank you!

**Comment by macropay-solutions** [comment]
Hi.
 We also use a lot of macros to extend the framework's behavior. The question/debate/issue is: overhead of builtin vs macros. And I think macros were implemented just for this. To allow extending the builtin functionality.

**Reply by stojkov-z** [reply]
Hi,
Thank you for your response. I agree that macros are a powerful way to extend Laravel without overloading the core. My proposal for `hasJoin` aims to provide a practical solution for common scenarios, but I understand the need to balance between core and macros. I'm interested to know if you consider `hasJoin` to be a common enough feature to justify its inclusion in the core, or if I should continue to implement it as a macro.

**Reply by macropay-solutions** [reply]
Any of us can't see the big picture as @taylorotwell sees it. 
We don't use/need hasJoin method for example. If it were to be added to the core, imagine that the builder is included in the Relation class and so also Relation will get it. 
There might be other implications that we don't know yet caused by the ForwardsCalls trait.
For this specific case those implications may do no harm...

**Reply by stojkov-z** [reply]
In our company's projects, `hasJoin` has often been useful and might make sense to include, but I understand your perspective regarding the implications. Thanks.
